src/main/java/com/google/sps/servlets/MemesServlet.java

```diff
@@ -87,5 +92,40 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       String imgTag = String.format("<img src=\"%s\" width = \"300\"/>", blob.getMediaLink());
       response.getWriter().println(imgTag);
     }
+  }
+
+/** This funtions  randomly  get a string  captions from an array of string. When the function is called
+ * it randomly return memes caption from an array
+ * input: null
+ * return: strings..
+ * 
+ */
+  
+
+  public static String getCpation(){
+//     initilaise an array  of captions
+        final String[] proper_noun = {"Fred", 
+        
+        "When the professor is passionate about teaching and you genuinely understan and enjoy the class.", 
+        " When ur best friend calls u and have some gossip to tell u.", 
+        "When you go off on him and later on realize you were actually wrong",
+        "When you tell everyone about your summer body goals, but you didn’t tell them which summer",
+
+        "  When you’ve been eating healthy for the past 15 minutes and STILL see no progress.",
+
+        
+    
+    };
+    //   import the random module
+        Random random = new Random();
+        // get random  index of array element
+        int index = random.nextInt(proper_noun.length);
+       // System.out.println(proper_noun[index]);
+
+    //    return random caption
+        return proper_noun[index];
+        
+        
```

Comment:
I don't know if it was your work, but lines 63 - 73 are a bit messy on the indentation (It's kind of in-line). It's a little confusing and hard to follow. Other than that, consider marking up the indentation a little bit more overall, I couldn't match all the brackets '{ }' to its pair as I was going through the code.